templates/includes/navbar_notifications.html.twig

```diff
@@ -1,22 +1,79 @@
-                        <span class="badge bg-{{ notice.type }} me-1"></span>
-                        {{ notice.message }}
-                    </a>
-                {% endfor %}
-            </div>
-        </div>
-    </div>
-{% endif %}
+{# @var notificationEvent \KevinPapst\TablerBundle\Event\NotificationEvent #}
+{% set notificationEvent = tabler_notifications() %}
+
+{% if notificationEvent.visible %}
+    <div class="nav-item dropdown d-none d-md-flex me-3">
+        {# Notification Icon #}
+        <a class="nav-link px-0" href="#"
+           data-bs-toggle="dropdown" tabindex="-1"
+           aria-label="{{ 'Show notifications'|trans({}, 'TablerBundle') }}"
+        >
+            <i class="icon far fa-bell"></i>
+            {% if notificationEvent.total > 0 -%}
+                <span class="badge bg-{{ notificationEvent.badgeColor }}">
+                {%- if notificationEvent.showBadgeTotal -%}
+                    {{ notificationEvent.total }}
+                {%- endif -%}
+            </span>
+            {%- endif %}
+        </a>
+
+        {# Dropdown #}
+        <div
+            class="dropdown-menu {% if notificationEvent.withArrow %}dropdown-menu-arrow{% endif %} dropdown-menu-end dropdown-menu-card"
+        >
+            <div class="card">
+                {# Header #}
+                {% if notificationEvent.titleHtml %}
+                    {{ notificationEvent.titleHtml | raw }}
+                {% else %}
+                    <div class="card-header">
+                        <div class="card-title text-truncate">
+                            {% if notificationEvent.total > 0 %}
+                                {{ notificationEvent.title ?? 'You have %count% notifications'|trans({'%count%': notificationEvent.total}, 'TablerBundle') }}
+                            {% else %}
+                                {{ notificationEvent.titleEmpty ?? 'You have no notification'|trans({}, 'TablerBundle') }}
+                            {% endif %}
+                        </div>
+                    </div>
+                {% endif %}
+
+                {# Notifications #}
+                {% if notificationEvent.total > 0 %}
+                    <div class="list-group list-group-flush list-group-hoverable">
+                        {% for notification in notificationEvent.notifications %}
+                            {% if notification.html ?? false %}
+                                {{ notification.message | raw }}
+                            {% else %}
+                                <div
+                                    class="list-group-item {% if notification.active ?? false %}active{% endif %} {% if notification.disabled ?? false %}disabled{% endif %}"
+                                >
+                                    <div class="row align-items-center">
+                                        {% if notification.withBadge ?? true %}

```

Comment:
Does that work if it is an "old NotificationInterface" but not a "NotificationModel" ?

Reply:
Yes every null-coalescing, fallback to the right operand if not exist / undefined / null.

> {# returns the value of foo if it is defined and not null, 'no' otherwise #}
{{ foo ?? 'no' }}

Reply:
Interesting, sound like a Twig exclusive feature I didn't knew about.